### ISSUE TITLE
`IndexFilters` focus becomes invisible when switching from filtering to default mode

### ISSUE THREAD

**Issue by kdaviduik** [Shopify/polaris-react-archive#12444]
### Summary

This seems to be a subtle [WCAG AA failure](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html). It also makes it a bit confusing/odd to understand when testing this using a screen reader - it announces "object replacement character" when this happens (using VoiceOver).

### Expected behavior

Upon switching from filtering to default mode, **the search button should become visibly focused.**

### Actual behavior

Upon switching from filtering to default mode, **there is no visibly focused element.** When navigating with a keyboard, the next focusable element is the checkbox inside the table. The previous (ie: with shift+tab) focusable element is the sort button.

https://github.com/user-attachments/assets/5fd48ad4-0fa4-44fd-9650-52a253a299d7

### Steps to reproduce

[Link to sandbox](https://codesandbox.io/p/sandbox/admiring-http-8hlgwq?file=%2Fsrc%2FApp.tsx%3A131%2C1)
1. Switch to filtering mode (pressing 'f' or clicking on the icon)
2. Switch to default mode (pressing escape, or clicking on the cancel button)
3. Observe that no element is visibly focused

### Are you using React components?

Yes

### Polaris version number

Polaris internal 13.18.0

### Browser

Chrome: 126.0.6478.183

### Device

Macbook Pro (running macOS 14.5)
